### PR TITLE
[iQue] Only strip debug sections when comparing objects

### PR DESCRIPTION
**Pull request by Thar0**
Re-matches any files that turned out to be non-matching due to symtab/reloc differences.

### CODE REVIEW

File: include/sys/asm.h [hunk 1]
```diff
@@ -22,49 +22,84 @@ modified versions thereof.
...
+    x:                     ;\
+    .frame  sp,0,ra
+    //.type   x, @function   ;\
 
-#define	END(proc)					\
-	.end	proc
+#if defined(BBPLAYER) || defined(__sgi)
+#define XLEAF(x)    \
+    .globl  x      ;\
+    x:
```
Comment (hensldm) [inline include/sys/asm.h:47]:
```suggestion
    .globl  x      ;\
    .aent   x,0;   ;\
    x:
```
Does egcs allow this? (this is needed for irix which you've added the define for)

File: include/sys/asm.h [hunk 2]
```diff
@@ -22,49 +22,84 @@ modified versions thereof.
...
-	x	=	y
-	
-#define	EXPORT(x)					\
-	.globl	x;					\
-x:
+#define EXPORT(x)   \
+    .globl  x      ;\
+    x:
 
 #ifdef BBPLAYER
```
Comment (hensldm) [inline include/sys/asm.h:70]:
```suggestion
#if defined(BBPLAYER) || defined(__sgi)
```
If you want, this looks to be exactly what irix uses.

File: include/sys/asm.h [hunk 3]
```diff
@@ -22,49 +22,84 @@ modified versions thereof.
...
 
-#define	XLEAF(x)					\
-	.global x;
+#define LEAF(x)             \
+    .globl  x              ;\
+    .align  2              ;\
+    .ent    x,0            ;\
+    x:                     ;\
+    .frame  sp,0,ra
+    //.type   x, @function   ;\
```
Comment (hensldm) [inline include/sys/asm.h:42]:
```suggestion
```
I'm guessing this didn't work out?

File: include/sys/asm.h (hunk 1 above)
Comment (Thar0) [reply include/sys/asm.h:47]: `aent` is theoretically supported by this version of gas but for some reason it claims it's unsupported. I guess I'll need to add another version of the macro for irix unless that's figured out

File: include/sys/asm.h (hunk 3 above)
Comment (Thar0) [reply include/sys/asm.h:42]: Somehow this works now lol, I guess some other change made it work

File: include/sys/asm.h [hunk 4]
```diff
@@ -22,49 +22,92 @@ modified versions thereof.
...
 
-#define	LEAF(x)						\
-	.globl	x;					\
-	.ent	x,0;					\
-x:;							\
-	.frame	sp,0,ra
 
-#define	XLEAF(x)					\
-	.global x;
+// libgultra doesn't match with the .type directive but iQue sdk asm.h uses it
```
Comment (hensldm) [inline include/sys/asm.h:36]:
```suggestion
/* libgultra doesn't match with the .type directive but iQue sdk asm.h uses it */
```
Ido is picky picky

File: include/sys/asm.h [hunk 5]
```diff
@@ -22,49 +22,92 @@ modified versions thereof.
...
+#define ASM_TYPE_FUNC(x)    .type   x, @function
+#else
+#define ASM_TYPE_FUNC(x)
+#endif
 
-#define	END(proc)					\
-	.end	proc
+#define LEAF(x)             \
+    .globl  x              ;\
+    .align  2              ;\
```
Comment (hensldm) [inline include/sys/asm.h:45]: This builds `libgultra_rom` fine for you right? Apparently with changes I've made, adding an `.align 2` for `libgultra_rom` mismatches on exceptasm.

**Review by hensldm — COMMENTED** [review]
Couple more things.

File: include/sys/asm.h (hunk 5 above)
Comment (Thar0) [reply include/sys/asm.h:45]: I double-checked libgultra_rom yeah, it all matches for me 🤔

**Review by hensldm — APPROVED** [review]

**Pull request merged**